tests/console/rust_cargo.pm

```diff
@@ -0,0 +1,84 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: cargo
+# Summary: Install cargo and assert full functionality
+#   - Create a new project
+#   - Add documentation
+#   - Add a dependency
+#   - Validate that the project can be run
+#   - Valdiate that documentation can be built by cargo
+# Maintainer: QE Core <qe-core@suse.de>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils qw(zypper_call);
+use Utils::Architectures 'is_aarch64';
+
+sub run {
+    select_console('root-console');
+    zypper_call('in cargo gcc');
+
+    # Setup test project
+    my $proj_name = "test_project";
+    my $timeout = (is_aarch64) ? 600 : 300;
+    my $test_arg = 'openQA';
+    select_console('user-console');
+    assert_script_run('cargo_new ' . $proj_name . " && cd " . $proj_name);
+    assert_script_run(qq(echo 'uuid = "0.8"' >> $proj_name/Cargo.toml));
```

Comment:
i'd rather use cargo add to handle dependencies

Reply:
fair enough, though manually adding dependencies is a valid use case and feature of the `Cargo.toml` format and cargo is capable of detecting dependencies that were manually added at build time.

Reply:
Added a separate test function which replaces this. Also adds the dependency in its current version.